src/time_bucket.c

```diff
@@ -459,6 +462,91 @@ ts_date_offset_bucket(PG_FUNCTION_ARGS)
 	PG_RETURN_DATUM(date);
 }
 
+TS_FUNCTION_INFO_V1(ts_uuid_bucket);
+
+Datum
+ts_uuid_bucket(PG_FUNCTION_ARGS)
+{
+	Datum ts = DirectFunctionCall1(ts_timestamptz_from_uuid_v7, PG_GETARG_DATUM(1));
+	Datum origin = (PG_NARGS() > 2 ? PG_GETARG_DATUM(2) : TimestampTzGetDatum(DEFAULT_ORIGIN));
+
+	PG_RETURN_DATUM(DirectFunctionCall3(ts_timestamptz_bucket, PG_GETARG_DATUM(0), ts, origin));
+}
+
+TS_FUNCTION_INFO_V1(ts_uuid_offset_bucket);
+
+TSDLLEXPORT Datum
+ts_uuid_offset_bucket(PG_FUNCTION_ARGS)
+{
+	if (PG_ARGISNULL(1))
+		PG_RETURN_NULL();
```

Comment:
Should we have this check in `ts_uuid_bucket()` as well?

Reply:
Actually, I realized that all time_bucket functions except one is defined STRICT, so they will return NULL on any NULL argument input. Therefore, only one function needs NULL checks. Updated code and tests to reflect that.